### ISSUE TITLE
High Duration Loss

### ISSUE THREAD

**Issue by w11wo** [mush42/optispeech#1]
Hi @mush42 , thank you so much for this wonderful repository. It's been a joy playing with it for the past few weeks. It's incredibly well written and structured!

I'd like to ask, what values of duration losses are you getting, or could you please share your Tensorboard results?

I've been trying a couple of different configs, but the duration loss values have always been quite high (>100) for some reason, and I'd like to make sure that this is the expected result:

![image](https://github.com/user-attachments/assets/aa21df2b-c730-4a5f-9dfa-29b21cad0afd)

Thanks in advance :)

**Comment by mush42** [comment]
Hi @w11wo 

I can confirm this issue, I experienced high `train_duration_loss`, but converging `val_duration_loss` so far. Despite this, the model learns to speak very early, around step 20,000, and all other losses converge as expected.

I believe this happens because the model is trying to align and learn to speak at the same time.

Some relevant suggestions are offered in [this issue](https://github.com/espnet/espnet/issues/4432#issuecomment-1149699330), namely, increasing `lambda_align` to 5.0, but I didn't try it yet.

Best
Musharraf

**Comment by mush42** [comment]
Also, what config values did you changed?

**Comment by w11wo** [comment]
Thanks for the response @mush42. I will try increasing the loss scale as you mentioned.

What I meant by trying different configs are just different encoder/decoder archs., e.g. Transformer, LightSpeech, ConvNext. But I think this shouldn't really affect it. I also used 44.1kHz in my experiments, but I've also tried 24kHz and it still had a similar duration loss.

So far the results are intelligible. I just wanted to make sure that this is expected.

**Comment by w11wo** [comment]
Hi @mush42, thought I'd share some results.

Here's my intermediate training results, at about 700k steps: https://voca.ro/1cpbiMltzrVq

I think changing `lambda_align = 5.0` helps with the overall quality of the generated audio. I find that it has better pauses and pacing. And I saw that you've made this value default in https://github.com/mush42/optispeech/commit/3152a8d9fd8f01b14940f054feb776f00eb279f2.

Just to be clear with configs, this is an Indonesian (`id`) speaker with the following feature extraction config:

```yaml
sample_rate: 44100
n_feats: 80
n_fft: 2048
hop_length: 512
win_length: 2048
f_min: 20
f_max: 11025
```

And the model is ConvNext, trained with 782 utterances.

Overall, I'm pretty satisfied with the audio quality given the RTF of this model. I used to use LightSpeech (TensorFlowTTS implementation), which is quite fast but the quality isn't the best. With OptiSpeech, I think I get the best of both worlds: fast inference speeds whilst maintaining the quality.

Great work, @mush42!

**Comment by mush42** [comment]
Thanks @w11wo for sharing. Sounds great!

**Comment by rmcpantoja** [comment]
Hi @w11wo, 
You said you trained using 782 utterances. Did you train from scratch or fine-tuned using an existing model?
Cheers.

**Comment by w11wo** [comment]
@rmcpantoja, I trained from scratch.
